content/tutorial/tutorial.md

```diff
@@ -12,185 +12,189 @@ redirect_from:
+`render` मेथड स्क्रीन पर जो आप देखना चाहते हैं उसका *वर्णन* देती है। React वर्णन लेती है और परिणाम प्रदर्शित करती है। विशेष रूप से, रेंडर एक **React एलिमेंट** देता है, जो की रेंडर का हल्का विवरण है। अधिकांश React डेवलपर्स "JSX" नामक एक विशेष सिंटेक्स का उपयोग करते हैं जो इन इसट्रकचरस को लिखना आसान बनाता है। `<div />` सिंटैक्स को बिल्ड समय में `React.createElement ('div')` में बदल दिया जाता है। ऊपर दिया गया उदाहरण इसके बराबर है:
 
 ```javascript
 return React.createElement('div', {className: 'shopping-list'},
-  React.createElement('h1', /* ... h1 children ... */),
-  React.createElement('ul', /* ... ul children ... */)
+  React.createElement('h1', /* ... h1 चिल्ड्रन ... */),
+  React.createElement('ul', /* ... ul चिल्ड्रन ... */)
 );
 ```
 
-[See full expanded version.](babel://tutorial-expanded-version)
+[पूर्ण विस्तारित संस्करण देखें।](babel://tutorial-expanded-version)
 
-If you're curious, `createElement()` is described in more detail in the [API reference](/docs/react-api.html#createelement), but we won't be using it in this tutorial. Instead, we will keep using JSX.
+यदि आप उत्सुक हैं, तो [एपीआई संदर्भ](/docs/react-api.html#createelement) में `createElement()` को अधिक वर्णन के साथ बताया गया है, लेकिन हम इस ट्यूटोरियल में इसका उपयोग नहीं करेंगे। इसके बजाय, हम JSX का उपयोग करते रहेंगे।
 
-JSX comes with the full power of JavaScript. You can put *any* JavaScript expressions within braces inside JSX. Each React element is a JavaScript object that you can store in a variable or pass around in your program.
+JSX जावास्क्रिप्ट की पूरी शक्ति के साथ आता है। आप JSX के अंदर ब्रेसिज़ के भीतर *कोई* जावास्क्रिप्ट एक्सप्रेशन रख सकते हैं।
+प्रत्येक React तत्व एक जावास्क्रिप्ट ऑब्जेक्ट है जिसे आप एक वैरिएबल् में स्टोर कर सकते हैं या अपने प्रोग्राम में चारों ओर से गुज़ार सकते हैं।
 
-The `ShoppingList` component above only renders built-in DOM components like `<div />` and `<li />`. But you can compose and render custom React components too. For example, we can now refer to the whole shopping list by writing `<ShoppingList />`. Each React component is encapsulated and can operate independently; this allows you to build complex UIs from simple components.
+`ShoppingList` कौम्पोनॅन्ट ऊपर केवल अंतर्निहित डोम जैसे `<div />` और `<li />` को रेंडर करता है। लेकिन आप कस्टम React कौम्पोनॅन्ट्स को भी कम्पोज़ और रेंडर कर सकते हैं। उदाहरण के लिए, हम अब पूरी खरीदारी सूची को `<ShoppingList />` लिखकर संदर्भित कर सकते हैं। प्रत्येक React कौम्पोनॅन्ट संकुचित है और स्वतंत्र रूप से काम कर सकता है; यह आपको सरल कौम्पोनॅन्टों से जटिल यूआई बनाने की अनुमति देता है।
```

Comment:
अंतर्निहित too complex

Reply:
रेंडर -> don't translate

Reply:
जटिल यूआई -> we had a discussion regarding this at some other point too?

Reply:
> अंतर्निहित too complex

`बिल्ट-इन` ?

Reply:
> रेंडर -> don't translate

I think we are not translating render if its referring to the render function, otherwise we are translating it.